tests/conftest.py

```diff
@@ -1,138 +1,532 @@
-import firecrest as f7t
+from aiida import orm
+import firecrest
+import firecrest.path
 import pytest
 import requests
-import yaml
 
-from aiida_firecrest.utils_test import FirecrestConfig, FirecrestMockServer
 
+class Values:
+    _DEFAULT_PAGE_SIZE: int = 25
 
-def pytest_report_header(config):
-    if config.getoption("--firecrest-config"):
-        header = [
-            "Running against FirecREST server: {}".format(
-                config.getoption("--firecrest-config")
-            )
-        ]
-        if config.getoption("--firecrest-no-clean"):
-            header.append("Not cleaning up FirecREST server after tests!")
-        return header
-    return ["Running against Mock FirecREST server"]
 
+@pytest.fixture
+def firecrest_computer(firecrest_config):
+    """Create and return a computer configured for Firecrest.
 
-def pytest_terminal_summary(
-    terminalreporter: TerminalReporter, exitstatus: int, config: pytest.Config
-):
-    """Called after all tests have run."""
-    data = config.stash.get("firecrest_requests", None)
-    if data is None:
-        return
-    terminalreporter.write(
-        yaml.dump(
-            {"Firecrest requests telemetry": data},
-            default_flow_style=False,
-            sort_keys=True,
-        )
+    Note, the computer is not stored in the database.
+    """
+
+    # create a temp directory and set it as the workdir
+
+    computer = orm.Computer(
+        label="test_computer",
+        description="test computer",
+        hostname="-",
+        workdir=firecrest_config.workdir,
+        transport_type="firecrest",
+        scheduler_type="firecrest",
+    )
+    computer.set_minimum_job_poll_interval(5)
+    computer.set_default_mpiprocs_per_machine(1)
+    computer.configure(
+        url=firecrest_config.url,
+        token_uri=firecrest_config.token_uri,
+        client_id=firecrest_config.client_id,
+        client_secret=firecrest_config.client_secret,
+        compute_resource=firecrest_config.compute_resource,
+        small_file_size_mb=firecrest_config.small_file_size_mb,
+        temp_directory=firecrest_config.temp_directory,
+        api_version=firecrest_config.api_version,
     )
+    return computer
+
+
+class MockFirecrest:
+    """Mocks py:class:`pyfirecrest.Firecrest`."""
+
+    def __init__(self, firecrest_url, *args, **kwargs):
+        self._firecrest_url = firecrest_url
+        self.args = args
+        self.kwargs = kwargs
+
+        self.whoami = whoami
+        self.list_files = list_files
+        self.stat = stat_
+        self.mkdir = mkdir
+        self.simple_delete = simple_delete
+        self.parameters = parameters
+        self.symlink = symlink
+        self.checksum = checksum
+        self.simple_download = simple_download
+        self.simple_upload = simple_upload
+        self.compress = compress
+        self.extract = extract
+        self.copy = copy
+        self.submit = submit
+        self.poll_active = poll_active
```

Comment:
This pattern is a bit weird for me. Cant you move the global functions you set here to the class as methods? So for example
```
class MockFirecrest:
    def __init__(self, firecrest_url, *args, **kwargs):
        self.whoami = whoami

def whoami(...):
    ...
```
->
```
class MockFirecrest:
    def __init__(self, firecrest_url, *args, **kwargs):
        ...
    def whoami(self, ...):
        ...
```

Reply:
If the need to be you can mark them as @staticmethod

Reply:
Thanks @agoscinski for your good suggestion.
I initially defined it this way just to remember what I have mocked and what I haven't mocked. 

But now that tests has been developed, we can do it the right way.